### PR TITLE
35 aid the user in importing dao interfaces into an application model

### LINKED ISSUE #35: Aid the user in importing DAO interfaces into an Application Model
After #34, in order to complete a model such as Fig. 9 of the [FrameWeb 2020 paper](http://www.inf.ufes.br/~vitorsouza/wp-content/papercite-data/pdf/souza-celebratingfalbo20.pdf), a user should bring to the diagram a persistence package created in another diagram, have this package contain only the required DAO interfaces and then associate the services to their respective DAO interface dependencies.

The FrameWeb plug-in should provide an easier way to do this, such as right-click a Service Class > Add DAO Dependency > Some Package Name > Some DAO Interface name. By selecting a DAO interface, the plug-in should:

- Verify if the package of this DAO interface is already in the diagram, if not add it;
- Verify if the DAO interface is already in the diagram, if not add it;
- Verify if the service class already has an association with the DAO interface, if not add it.

Similar issue, but for controller dependencies: #47 .

### PR DESCRIPTION
**Pull request by gabrielgatti7**
Closes #35

### CODE REVIEW

File: src/main/java/br/ufes/inf/nemo/frameweb/vp/FrameWebPlugin.java [hunk 1]
```diff
@@ -69,6 +69,9 @@ public class FrameWebPlugin implements VPPlugin {
   /** Indicates if the Plug-in Settings Dialog is open. */
   private boolean pluginSettingsDialogOpen = false;
 
+  /** Indicates if the Select DAO Dialog is open. */
+  private boolean SelectDAODialogOpen = false;
```
Comment (vitorsouza) [inline src/main/java/br/ufes/inf/nemo/frameweb/vp/FrameWebPlugin.java:73]:
```suggestion
  private boolean selectDAODialogOpen = false;
```

(Coding standards, attributes begin with lower case)

File: src/main/java/br/ufes/inf/nemo/frameweb/vp/FrameWebPlugin.java [hunk 2]
```diff
@@ -92,6 +95,14 @@ public void setPluginSettingsDialogOpen(boolean pluginSettingsDialogOpen) {
...
   }
 
+  public boolean isSelectDAODialogOpen() {
+    return SelectDAODialogOpen;
+  }
+
+  public void setSelectDAODialogOpen(boolean SelectDAODialogOpen) {
+    this.SelectDAODialogOpen = SelectDAODialogOpen;
+  }
+
```
Comment (vitorsouza) [inline src/main/java/br/ufes/inf/nemo/frameweb/vp/FrameWebPlugin.java:105]:
```suggestion
  public boolean isSelectDAODialogOpen() {
    return selectDAODialogOpen;
  }

  public void setSelectDAODialogOpen(boolean selectDAODialogOpen) {
    this.selectDAODialogOpen = selectDAODialogOpen;
  }

```

File: src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java [hunk 3]
```diff
@@ -24,11 +38,38 @@
  * @version 0.0.1
  */
 public class AddFrameWebDependencyToClassContextController implements VPContextActionController {
+    /** The contents of the Add FrameWeb Dependency dialog. */
+    private SelectDAOPanel selectDAOPanel;
+
+    /** The Plug-in Settings dialog. */
```
Comment (vitorsouza) [inline src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java:44]:
```suggestion
    /** The Add FrameWeb Dependency dialog. */
```

File: src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java [hunk 4]
```diff
@@ -24,11 +38,38 @@
...
+    /** List of all DAO Interfaces that have an association with the service class */
+    private List<IModelElement> associatedDAOInterfaces;
+    /** Application packages in the diagram*/
+    private List<IDiagramElement> applicationPackages;
+
+
     /** Called when the button is pressed. Adds the dependency to the class and diagram. */
     @Override
     public void performAction(final VPAction action, final VPContext context, final ActionEvent event) {
+        Logger.log(Level.CONFIG, "Performing action: Add FrameWeb Constraint (Class Attribute) > {0}",
```
Comment (vitorsouza) [inline src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java:69]:
```suggestion
        Logger.log(Level.CONFIG, "Performing action: Add FrameWeb Dependency (Class) > {0}",
```

Aren't we adding a FrameWeb depedency to a class? Why Add FrameWeb Constraint (Class Attribute)? Am I missing something?

File: src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java [hunk 5]
```diff
@@ -40,42 +81,109 @@ public void performAction(final VPAction action, final VPContext context, final
...
-        // logs the classes that will be added as dependencies
-        for (final IClass clazz : frameWebClasses) {
-            final IModelElement clazzPackage = clazz.getParent();
+        // Create a map that associates the name of the classes (key) to their corresponding IClasses (value)
+        this.daoClassesMap = new HashMap<>();
 
-            Logger.log(Level.INFO, "Adding dependency {0} - {1} to {2}",
-                    new Object[] { clazz.getName(), clazzPackage.getName(), modelElement.getName() });
+        // Checks if the DAO Interface is already in the diagram, if it has association with the service class and
+        // if it's parent (persistence package) is in the diagram. If one of these conditions is false, adds the DAO
```
Comment (vitorsouza) [inline src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java:135]:
```suggestion
        // if its parent (persistence package) is in the diagram. If one of these conditions is false, adds the DAO
```

**Review by vitorsouza — CHANGES_REQUESTED** [review]
I suggest some minor fixes.

File: src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java (hunk 4 above)
Comment (gabrielgatti7) [reply src/main/java/br/ufes/inf/nemo/frameweb/vp/controllers/AddFrameWebDependencyToClassContextController.java:69]: My mistake. I copied this line from another class and forgot to change.

**Review by vitorsouza — APPROVED** [review]

**Pull request merged**